packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Square, DotsThreeVertical, X } from '@phosphor-icons/react';
+import React, { type HTMLAttributes, type KeyboardEvent, useEffect, useRef, useState } from 'react';
+
+import { Button, DropdownMenu, Input } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { getNext, getParent, getPrevious, getItems, type Item, getLastDescendent } from './types';
+
+// TODO(burdon): Break/join lines.
+// TODO(burdon): TextObject/MarkdownEditor
+// TODO(burdon): Cut-and-Paste.
+// TODO(burdon): Drag-and-drop.
+// TODO(burdon): Better key nav.
+
+type TreeOptions = Pick<HTMLAttributes<HTMLInputElement>, 'placeholder' | 'spellCheck'> & {
+  checkbox?: boolean;
+};
+
+//
+// Item
+//
+
+type ItemProps = {
+  item: Item;
+  active?: boolean;
+  onFocus?: () => void;
+  onEnter?: (before?: boolean) => void;
+  onDelete?: () => void;
+  onIndent?: (left?: boolean) => void;
+  onNav?: (item: Item, direction?: 'home' | 'end' | 'up' | 'down') => void;
+} & TreeOptions;
+
+const Item = ({
+  item,
+  active,
+  checkbox,
+  placeholder,
+  spellCheck = false,
+  onFocus,
+  onEnter,
+  onDelete,
+  onIndent,
+  onNav,
+}: ItemProps) => {
+  const [focused, setFocused] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  useEffect(() => {
+    if (active) {
+      inputRef.current?.focus();
+    }
+  }, [active]);
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    switch (event.key) {
+      case 'ArrowUp':
+        // TODO(burdon): Maintain caret position.
+        // TODO(burdon): Go to first/last child of group.
+        onNav?.(item, event.shiftKey ? 'home' : 'up');
+        break;
+      case 'ArrowDown':
+        onNav?.(item, event.shiftKey ? 'end' : 'down');
+        break;
+      case 'Tab':
+        event.preventDefault();
+        onIndent?.(event.shiftKey);
+        break;
+      case 'Enter':
+        // TODO(burdon): Split line (shift to create new).
+        onEnter?.(!!item.text?.length && inputRef.current?.selectionStart === 0);
+        break;
+      case 'Backspace':
+        // TODO(burdon): Merge with previous if caret at start.
+        if (!item.text?.length) {
+          event.preventDefault();
+          onDelete?.();
+        }
+        break;
+      case 'Escape':
+        // TODO(burdon): Toggle checkbox?
+        break;
+    }
+  };
+
+  return (
+    <div className='flex items-center px-2 gap-3'>
```

Comment:
Did `TreeItem.Root` not work for this?
If not, this should have `role='treeitem'`

Reply:
Is `flex` necessary? Does the layout work as expected when the content is long?

Reply:
Perhaps I could have used TreeItem more extensively throughout?
This whole thing started as a devtools quick experiment. Let's discuss.